src/main/scala/org/clulab/wm/eidos/exporters/DebugGroundingExporter.scala

```diff
@@ -1,41 +1,175 @@
-        val args = em.eidosArguments("cause") ++ em.eidosArguments("effect")
-        val info = args.map(m => (m.odinMention.text, m.grounding.get("wm_compositional"), m.grounding.get("wm_flattened")))
-        for ((text, groundingComp, groundingFlat) <- info) {
-          pw.println(s"text: ${text}")
-          pw.println(s"Compositional Grounding:")
-          groundingComp match {
-            case None => pw.println("no grounding...")
-            case Some(grounding) =>
-              grounding.grounding.foreach { gr =>
-                pw.println(s"  --> grounding: ${gr.name}")
-                pw.println(s"      score: ${gr.score}")
+
+//      pw.println("********************************************")
+      val doc = annotatedDocument.document
+      for (i <- doc.sentences.indices) {
+        val clusent = reader.components.procOpt.get.asInstanceOf[EidosCluProcessor].annotate(doc.sentences(i)
+          .getSentenceText).sentences.head
+        pw.println("********************************************\n")
+        pw.println(s"Sentence $i: ${clusent.getSentenceText}.\n")
+        pw.println("SRLS:")
+        pw.println(clusent.enhancedSemanticRoles.getOrElse(None))
+//        pw.println()
+        pw.println("DEPS:")
+        pw.println(clusent.dependencies.get)
+        val mentions = annotatedDocument.eidosMentions.filter(_.odinMention.sentence == i)
+
+        mentions.filter(_.odinMention matches "Causal").foreach { em =>
+          val args = em.eidosArguments("cause") ++ em.eidosArguments("effect")
+          val info = args.map(m => (m.odinMention.text, m.grounding.get("wm_compositional"), m.grounding.get("wm_flattened")))
+          for ((text, groundingComp, groundingFlat) <- info) {
+            if (text=="famine") {
+              //do nothing
+            }
+            else {
+              pw.println(s"mention text: ${text}\n")
+              pw.println(s"mention entities: ${args.head.odinMention.entities.get}\t")
+
+              pw.println(s"Flat Grounding:")
+              groundingFlat match {
+                case None => pw.println("no grounding...")
+                case Some(grounding) =>
+                  grounding.grounding.foreach { gr =>
+                    pw.println(s"  --> grounding:  ${gr.name}")
+                    pw.println(s"      score:      ${gr.score}")
+                  }
               }
-          }
-          pw.println(s"Flat Grounding:")
-          groundingFlat match {
-            case None => pw.println("no grounding...")
-            case Some(grounding) =>
-              grounding.grounding.foreach { gr =>
-                pw.println(s"  --> grounding: ${gr.name}")
-                pw.println(s"      score: ${gr.score}")
+              pw.println()
+
+              pw.println(s"Compositional Grounding:")
+              groundingComp match {
+                case None => pw.println("no grounding...")
+                case Some(grounding) =>
+                  var tupNum: Int = 0
```

Comment:
It's not important, but there is a utils.Counter in eidoscommon that might be used.